frontend/packages/client/src/components/Community/CommunityPropsAndVoting.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import ActionButton from 'components/ActionButton';
+import StrategySelectorForm from 'components/Community/StrategySelectorForm';
+import isEqual from 'lodash/isEqual';
+
+// this object is used to change
+// field names as are used
+// on the backend
+const fieldMapPayload = {
+  contractAddress: 'addr',
+  contractName: 'name',
+  maxWeight: 'maxWeight',
+  minimunBalance: 'threshold',
+  publicPath: 'publicPath',
+};
+
+// this function renames fields if necessary to prepare payload for backend
+const mapFieldsForBackend = (contract) => {
+  return Object.assign(
+    {},
+    ...Object.entries(contract).map(([key, value]) => ({
+      ...(fieldMapPayload[key] && value
+        ? {
+            [fieldMapPayload[key]]: value,
+          }
+        : value
+        ? { [key]: value }
+        : undefined),
+    }))
+  );
+};
+
+const hasListChanged = (newList, originalList) => {
```

Comment:
This would seem like it can be directly used in the `enabled` prop on line 64, rather than a separate function.

Reply:
removed